model/src/pyrenew/latent/hospitaladmissions.py

```diff
@@ -16,54 +15,153 @@
+    ) -> None:
+        """Default constructor
+
+        Parameters
+        ----------
+        dist : dist.Distribution, optional
+            Prior distribution of the IHR, by default
+            dist.LogNormal(jnp.log(0.05), 0.05)
+        varname : str, optional
+            Name of the random_variable that may hold observed IHR, by default
+            "IHR"
+
+        Returns
+        -------
+        None
+        """
+
+        self.validate(dist)
+
+        self.dist = dist
+        self.varname = varname
+
+        return None
+
+    @staticmethod
+    def validate(distr: dist.Distribution) -> None:
+        assert isinstance(distr, dist.Distribution)
+
+    def sample(
+        self,
+        random_variables: dict = None,
+        constants: dict = None,
+    ) -> InfectHospRateSample:
+        return InfectHospRateSample(
+            npro.sample(
+                "IHR",
+                self.dist,
+                obs=random_variables.get(self.varname, None),
+            )
+        )
+
 
 class HospitalAdmissions(RandomVariable):
-    """Latent hospital admissions"""
+    r"""Latent hospital admissions
+
+    Implements a renewal process for the expected number of hospitalizations.
+
+    Notes
+    -----
+
+    The following text was directly extracted from the wastewater model
+    documentation
+    (`link <https://github.com/cdcent/cfa-forecast-renewal-ww/blob/a17efc090b2ffbc7bc11bdd9eec5198d6bcf7322/model_definition.md#hospital-admissions-component> `_).
+
+    Following other semi-mechanistic renewal frameworks, we model the _expected_
+    hospital admissions per capita :math:`H(t)` as a convolution of the
+    _expected_ latent incident infections per capita :math:`I(t)`, and a
+    discrete infection to hospitalization distribution :math:`d(\tau)`, scaled
+    by the probability of being hospitalized :math:`p_\mathrm{hosp}(t)`.
+
+    To account for day-of-week effects in hospital reporting, we use an
+    estimated _weekday effect_ :math:`\omega(t)`. If :math:`t` and :math:`t'`
+    are the same day of the week, :math:`\omega(t) = \omega(t')`. The seven
+    values that :math:`\omega(t)` takes on are constrained to have mean 1.
+
+    .. math::
+
+        H(t) = \omega(t) p_\mathrm{hosp}(t) \sum_{\\tau = 0}^{T_d} d(\tau) I(t-\tau)
+
+    Where :math:`T_d` is the maximum delay from infection to hospitalization
+    that we consider.
+    """
 
     def __init__(
         self,
-        inf_hosp_int: ArrayLike,
+        infection_to_admission_interval: RandomVariable,
+        infect_hosp_rate_dist: RandomVariable,
         infections_varname: str = "infections",
-        IHR_obs_varname: str = "IHR",
         hospitalizations_predicted_varname: str = "predicted_hospitalizations",
-        IHR_dist: dist.Distribution = dist.LogNormal(jnp.log(0.05), 0.05),
+        weekday_effect_dist: RandomVariable = DeterministicVariable((1,)),
+        p_hosp_dist: RandomVariable = DeterministicVariable((1,)),
```

Comment:
Since this represents a reporting rate/probability, suggest renaming it to `p_report_dist` throughout.

Reply:
Addressed in 33d9ed0